src/TimeSelect.jsx

```diff
@@ -111,10 +141,67 @@ var TimePicker = React.createClass({
     return this.generateFormattedTime((this.props.value.getHours() * 100) + this.props.value.getMinutes());
   },
 
-  render: function() {
+  hoursChanged: function(e) {
+    this.props.minutesHoursChanged({ hours: e.target.value });
+  },
+
+  minutesChanged: function(e) {
+    this.props.minutesHoursChanged({ minutes: e.target.value });
+  },
+
+  generateTimeInput: function(timeUnit) {
+    var timeValues;
+    var label;
+    var currentTime;
+    var changeFunction;
+
+    if (timeUnit === 'minutes') {
+      timeValues = this.generateMinutes();
+      label = 'Minutes';
+      currentTime = this.props.time.minutes;
+      changeFunction = this.minutesChanged;
+    }
+
+    if (timeUnit === 'hours') {
+      timeValues = this.generateHours();
+      label = 'Hours';
+      currentTime = this.props.time.hours;
+      changeFunction = this.hoursChanged;
+    }
 
     return (
-      <ReactIntl.IntlProvider locale={this.props.locale}>
+      <div className="col-xs-6">
+        <Input
+          type="select"
+          value={currentTime}
+          name={this.props.name}
+          className={this.props.className}
+          label={label}
+          onChange={changeFunction}
+          id={this.props.id}
+        >
+          {timeValues.map(timeData => {
+            return (
+              <option value={timeData}>{timeData}</option>
+            );
+          })}
+        </Input>
+      </div>
+    );
+  },
+
+  render: function() {
+    var timeInput = <span />;
```

Comment:
i dont think you need to define the timeInput with a span. var timeInput; should do.